src/mmio_types/fifo_enable.rs

```diff
@@ -0,0 +1,9 @@
+use super::*;
+
+#[derive(Debug, Clone, Copy, Default, PartialEq, Eq)]
+#[repr(transparent)]
+pub struct FifoEnable(u32);
```

Comment:
I think this should be just a u8, which saves on size if this is placed into a struct.

I also think we should have entries for the Sound ON checking bits. Even though they're read only, a person still might want to read them of course.

Reply:
Looking at how you handled other stuff, I split the same register into `FifoEnable` (which is R/W) and `SoundStatus` (which is read-only). Do you think it'd be better to just have a single R/W register definition that handles both?

Reply:
I'll accept the PR as-is for now, and we can have a look at this later. 

Opened: https://github.com/rust-console/gba/issues/148